dagrunner/plugin_framework.py

```diff
@@ -88,35 +193,53 @@ def __call__(
         time_taken = indx = patterns_found = files_found = 0
         fpaths_found = []
         file_count = len(args) if file_count is None else max(file_count, len(args))
-        while (
-            patterns_found < len(args)
-            and files_found < file_count
-            and time_taken < timeout
-        ):
-            fpattern = args[indx]
-            expanded_paths = glob(fpattern)
+
+        args = list(map(process_path, args))
+        while time_taken < timeout:
+            pattern = args[indx]
+            host = None
+            if ":" in pattern:
+                host, pattern = pattern.split(":")
+
+            if host:
+                # bash equivalent to python glob (glob on remote host)
+                expanded_paths = (
+                    subprocess.run(
+                        f"ssh {host} \"printf '%s\n' {pattern} | grep -v '*'\" || true",
+                        shell=True,
+                        check=True,
+                        text=True,
+                        capture_output=True,
+                    )
```

Comment:
Similar to above - maybe a loop of 2 or 3 attempts to account for connection glitches.